Update Markdown content docs for config rework (#2280)
withastro/docs: src/pages/en/guides/markdown-content.mdx

export default {
  markdown: {
    // Applied to .md and .mdx files
    remarkPlugins: [remarkToc],
    // Preserves remark-gfm and remark-smartypants
    extendDefaultPlugins: true,
    remarkPlugins: [remarkToc, rehypeAccessibleEmojis],
  },
  integrations: [mdx({
    // Applied to .mdx files only

>     rehypePlugins: [rehypeAccessibleEmojis],

Author: Note: if this were `remarkPlugins`, MDX would blow away existing Markdown plugins and require the user to reapply them. Removing this MDX extension example to leave explanation to a separate section.

</html>
```

### Extending Markdown config from MDX

> Astro's MDX integration will extend [your project's existing Markdown configuration](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) by default. To override individual options, you can specify their equivalent in your MDX configuration.

Reviewer: ```suggestion
Astro's MDX integration will extend [your project's existing Markdown configuration](/en/reference/configuration-reference/#markdown-options) by default. To override individual options, you can specify their equivalent in your MDX configuration.
```

> For example, say you need to disable GitHub-Flavored Markdown and apply a different set of remark plugins for MDX files. You can apply these options like so:

Reviewer: ```suggestion
The following example disables GitHub-Flavored Markdown and applies a different set of remark plugins for MDX files:
```

```ts
// astro.config.mjs
import { defineConfig } froAm 'astro/config';
import mdx from '@astrojs/mdx';

export default defineConfig({
  markdown: {
    syntaxHighlight: 'prism',
    remarkPlugins: [remarkPlugin1],
    githubFlavoredMarkdown: true,
  },
  integrations: [
    mdx({
      // `syntaxHighlight` inherited from Markdown

      // Markdown `remarkPlugins` ignored,
      // only `remarkPlugin2` applied.
      remarkPlugins: [remarkPlugin2],
      // `githubFlavoredMarkdown` overridden to `false`
      githubFlavoredMarkdown: false,
    })
  ]
});
```

> You may also prefer to avoid extending your Markdown config from MDX. For this, you can set [the `extendMarkdownConfig` option](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) (enabled by default) to `false`:

Reviewer: ```suggestion
To avoid extending your Markdown config from MDX, set [the `extendMarkdownConfig` option](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) (enabled by default) to `false`:
```

> You may also prefer to disable Markdown config extension. For this, you can set [the `extendMarkdownConfig` option](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) option to `false`:

Reviewer: My only issue here is making it explicit that you're only disabling Markdown options for MDX files. (Not disabling Markdown for Markdown)

Author: I think I made this better?
https://github.com/withastro/docs/pull/2280/commits/9f13d49c3cda1d0285afee6a6d32d5b53a6a3265

</html>
```

### Extending Markdown config from MDX

By default, Astro's MDX integration will extend [your project's existing Markdown configuration](https://docs.astro.build/en/reference/configuration-reference/#markdown-options). To override individual options, you can specify their equivalent in your MDX configuration.

> For example, say you need to disable GitHub-Flavored Markdown and apply a different set of remark plugins for MDX files. You can apply these options like so, with [the `extendMarkdownConfig` option](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) enabled by default:

Reviewer: ```suggestion
For example, to disable GitHub-Flavored Markdown and apply a different set of remark plugins for MDX files only:
```

```ts
// astro.config.mjs
import { defineConfig } from 'astro/config';
import mdx from '@astrojs/mdx';

export default defineConfig({
  markdown: {
    syntaxHighlight: 'prism',
    remarkPlugins: [remarkPlugin1],
    githubFlavoredMarkdown: true,
  },
  integrations: [
    mdx({
      // `syntaxHighlight` inherited from Markdown

      // Markdown `remarkPlugins` ignored,
      // only `remarkPlugin2` applied.
      remarkPlugins: [remarkPlugin2],
      // `githubFlavoredMarkdown` overridden to `false`
      githubFlavoredMarkdown: false,
    })
  ]
});
```

> You may also need to disable `markdown` config extension in MDX. For this, set `extendMarkdownConfig` to `false`:

Reviewer: ```suggestion
You may also need to disable your `markdown` config settings in MDX. For this, set [the `extendMarkdownConfig` option (enabled by default)](/en/guides/integrations-guide/mdx/#extendmarkdownconfig) to `false`:
```

Author: I think "disable your Markdown config settings" is a bit confusing. Ex. does that disable syntax highlighting?
